src/main/java/org/alfresco/heartbeat/InfoDataCollector.java

```diff
@@ -95,12 +130,24 @@ public void setDeploymentMethodProvider(DeploymentMethodProvider deploymentMetho
         this.deploymentMethodProvider = deploymentMethodProvider;
     }
 
+    public void setDataSource(DataSource dataSource)
+    {
+        this.dataSource = dataSource;
+    }
+
+    @Override public void setServletContext(ServletContext servletContext)
+    {
+        this.servletContext = servletContext;
+    }
+    
     @Override
     public void afterPropertiesSet() throws Exception
     {
         PropertyCheck.mandatory(this, "serverDescriptorDAO", serverDescriptorDAO);
         PropertyCheck.mandatory(this, "currentRepoDescriptorDAO", currentRepoDescriptorDAO);
         PropertyCheck.mandatory(this, "deploymentMethodProvider", deploymentMethodProvider);
+        PropertyCheck.mandatory(this, "dataSource", dataSource);
+        PropertyCheck.mandatory(this, "servletContext", servletContext);
```

Comment:
Don't check on the `servlet context`, this will be null in the tests, since tests start with a minimal context, if any.
This is the cause of the failing build btw.

Reply:
Removed.